src/views/Simulation/components/Charts/TimelineControls.js

```diff
@@ -107,59 +133,40 @@ const TimelineControls = ({ chartData, markers, startDate, endDate }) => {
       isPlayingRef.current = true;
       startTimeRef.current = performance.now() - (pausedAtRef.current || 0);
       pausedAtRef.current = 0;
-      animationFrameId.current = requestAnimationFrame(animate); // start loop
-    }
-  };
-
-  const stepDuration = 500; // ms per step at 1x speed
-
-  const animate = (now) => {
-    if (!isPlayingRef.current) return; // check live ref
-
-    const totalSteps = chartData.length - 1;
-    const elapsed = now - startTimeRef.current;
-    const step = Math.min(Math.floor(elapsed / (stepDuration / playbackSpeed)), totalSteps);
-
-    setCurrentTimestep(step);
-
-    if (step < totalSteps) {
       animationFrameId.current = requestAnimationFrame(animate);
-    } else {
-      setIsPlaying(false);
-      isPlayingRef.current = false;
-      setCurrentTimestep(0);
-      pausedAtRef.current = 0;
     }
-  };
-
-  useEffect(() => {
-    return () => cancelAnimationFrame(animationFrameId.current);
-  }, []);
+  }, [animate]);
 
-  const handleSpeedChange = () => {
-    const speeds = [0.5, 1, 2];
-    const currentIndex = speeds.indexOf(playbackSpeed);
-    const nextSpeed = speeds[(currentIndex + 1) % speeds.length];
+  const handleSpeedChange = useCallback(() => {
+    const currentIndex = SPEEDS.indexOf(playbackSpeed);
+    const nextSpeed = SPEEDS[(currentIndex + 1) % SPEEDS.length];
     setPlaybackSpeed(nextSpeed);
 
     if (isPlayingRef.current) {
       const elapsed = performance.now() - startTimeRef.current;
       startTimeRef.current = performance.now() - elapsed * (playbackSpeed / nextSpeed);
     }
-  };
+  }, [playbackSpeed]);
 
-  const handleChartClick = (e) => {
-    if (e?.activeLabel != null) {
-      const newStep = e.activeLabel;
-      setCurrentTimestep(newStep);
+  const handleChartClick = useCallback(
+    (e) => {
```

Comment:
Using variable names as short as `e` will probably decrease readability for the other developers :)
We can usually find short and understandable names (e.g. `event`), and in other cases, we'd rather have a long variable (or function) name than one that is too short and could cause confusion :smile: